test/e2e/suite/issuers/acme/issuer.go

```diff
@@ -267,4 +273,76 @@ var _ = framework.CertManagerDescribe("ACME Issuer", func() {
 			})
 		Expect(err).NotTo(HaveOccurred())
 	})
+	It("ACME account with External Account Binding", func() {
+
+		By("providing the legacy keyAlgorithm value")
+
+		var (
+			secretName = "test-secret"
+			keyID      = "kid-1"
+			key        = "kid-secret-1"
+		)
+
+		// TODO: this value will get base64 encoded twice. Investigate why we need to do this.
+		keyBytes := []byte(base64.RawURLEncoding.EncodeToString([]byte(key)))
+		s := gen.Secret(secretName,
+			gen.SetSecretNamespace(f.Namespace.Name),
+			gen.SetSecretData(map[string][]byte{
+				"key": keyBytes,
+			}))
+		_, err := f.KubeClientSet.CoreV1().Secrets(f.Namespace.Name).Create(context.TODO(), s, metav1.CreateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		acmeIssuer := gen.Issuer(issuerName,
+			gen.SetIssuerNamespace(f.Namespace.Name),
+			gen.SetIssuerACMEEmail(testingACMEEmail),
+			gen.SetIssuerACMEURL(f.Config.Addons.ACMEServer.URL),
+			gen.SetIssuerACMESkipTLSVerify(true),
+			gen.SetIssuerACMEPrivKeyRef(testingACMEPrivateKey),
+			gen.SetIssuerACMEEABWithKeyAlgorithm(keyID, secretName, cmacme.HS256))
+
+		acmeIssuer, err = f.CertManagerClientSet.CertmanagerV1().Issuers(f.Namespace.Name).Create(
+			context.TODO(), acmeIssuer, metav1.CreateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		err = util.WaitForIssuerCondition(f.CertManagerClientSet.CertmanagerV1().Issuers(f.Namespace.Name),
+			acmeIssuer.Name,
+			v1.IssuerCondition{
+				Type:   v1.IssuerConditionReady,
+				Status: cmmeta.ConditionTrue,
+			})
+		Expect(err).NotTo(HaveOccurred())
+
+		By("removing the legacy keyAlgorithm value")
+
+		acmeIssuer, err = f.CertManagerClientSet.CertmanagerV1().Issuers(f.Namespace.Name).Get(context.TODO(), acmeIssuer.Name, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		acmeIssuer = gen.IssuerFrom(acmeIssuer,
+			gen.SetIssuerACMEEAB(keyID, secretName))
+
+		_, err = f.CertManagerClientSet.CertmanagerV1().Issuers(f.Namespace.Name).Update(context.TODO(), acmeIssuer, metav1.UpdateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		// TODO: we should use observedGeneration here, but currently it won't
+		// be incremented correctly in this scenario.
+		// Verify that Issuer's Ready condition remains True for 5 seconds.
+		err = wait.Poll(time.Millisecond*200, time.Second*5, func() (bool, error) {
+			iss, err := f.CertManagerClientSet.CertmanagerV1().Issuers(f.Namespace.Name).Get(
+				context.TODO(), issuerName, metav1.GetOptions{})
+			if err != nil {
+				return false, err
+			}
+			if !apiutil.IssuerHasCondition(iss, v1.IssuerCondition{
+				Type:   v1.IssuerConditionReady,
+				Status: cmmeta.ConditionTrue,
+			}) {
+				return false, errors.New("expected Ready condition to be true, got false")
+			}
+			// keep polling
+			return false, nil
+		})
+		Expect(err).To(HaveOccurred())
+		Expect(err).To(MatchError(wait.ErrWaitTimeout))
+	})
```

Comment:
I'm not convinced that we need an E2E test for this. AFAICS it demonstrates that we can include or omit the KeyAlgorithm in our API calls and that the Issuer controller marks the Issuer Ready regardless....but it doesn't really tell us anything about whether the KeyAlgorithm is used.

I was going to suggest adding a unit-test instead, for the `Setup` function (above) but I see that it's over 200 lines long and has no existing unit-tests, so that might be a big rabbit hole.

Up to you. Remove this if you agree or leave it if you think it adds value.

Reply:
+1 to Richard's comments - I'm not sure if the e2e test provides much value (and I think a unit test would actually provide _more_ value) - but can appreciate this is a relatively untested area as is 😬

Reply:
I added that bit to the e2e tests because this PR changes the part that e2e-tests ACME Issuer with ExternalAccountBinding-  we are now only testing with an Issuer [without the keyAlgorithm field](https://github.com/jetstack/cert-manager/blob/master/test/e2e/suite/conformance/certificates/acme/acme.go#L43).

Because of that I thought it would be good to have a basic high level e2e test that verifies that 1) an Issuer with that field _can_ be created (gets past webhooks, becomes Ready) and 2) the field can be removed- which are the two things users probably would be doing?

I could look at unit testing the `Setup` function. I expect that it will need some refactoring so it might be a separate PR (that could come before this PR).

Reply:
Seems reasonable to keep in order to verify that it all works fine with and without! 😊